mysite/announcements/admin.py

```diff
@@ -0,0 +1,17 @@
+from django.contrib import admin
+from .models import Announcements
+from django.contrib.auth.models import User
+
+class AnnouncementsAdmin(admin.ModelAdmin):
+    list_display = ('title', 'message', 'date_time_expire')
+    change_form_template = "announcements/announcements_change_form.html"
+
+    def response_change(self, request, obj):
+        if "_send_announcement" in request.POST:
+            users = User.objects.all()
```

Comment:
- What is SQL query this ORM will do to fetch users from Database?
- Can we optimize this?
- Do we need all of this info?

Reply:
Select * from user
Rather than fetching the whole QuerySet we can fetch only required columns here.